### PR TITLE
Azurecompute - additional templateOptions for WindowsProvisioningConfiguration

### PR DESCRIPTION
**Pull request by bostko**
Listener protocol flag allowing https usage.
https://msdn.microsoft.com/en-us/library/azure/jj157194.aspx#Anchor_7

### CODE REVIEW

**Comment by bostko** [comment]
mvn build seems to be successful however jenkins report it failed.

```
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5:23.302s
[INFO] Finished at: Sat Mar 18 03:42:33 EDT 2017
....
...
[JENKINS] Archiving /scratch/jenkins/workspace/jclouds-labs-pull-requests/h2-jdbc/target/h2-jdbc-2.1.0-SNAPSHOT-tests.jar to org.apache.jclouds.labs/h2-jdbc/2.1.0-SNAPSHOT/h2-jdbc-2.1.0-SNAPSHOT-tests.jar
channel stopped
Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
Started calculate disk usage of workspace
Finished Calculation of disk usage of workspace in 0 seconds
Setting status of 4efa74f886442756b4d1c6a3e0ff02dd9c82d009 to FAILURE with url https://jclouds.ci.cloudbees.com/job/jclouds-labs-pull-requests/1708/ and message: 'Build finished. '
```

**Comment by andreaturli** [comment]
rebuild please

File: azurecompute/src/main/java/org/jclouds/azurecompute/compute/options/AzureComputeTemplateOptions.java [hunk 1]
```diff
@@ -54,6 +54,7 @@
    protected String networkSecurityGroupName;
    protected String reservedIPName;
    protected Boolean provisionGuestAgent;
+   protected Boolean winrmUseHttps;
```
Comment (andreaturli) [inline azurecompute/src/main/java/org/jclouds/azurecompute/compute/options/AzureComputeTemplateOptions.java]: isn't `String listenerProtocol` a better model ? we can then add a `ListenerProtocol` enum like:
```
   public enum ListenerProtocol {

      HTTP("Http"),
      HTTPS("Https");

      private String value;

      ListenerProtocol(String value) {
         this.value = value;
      }

      public static ListenerProtocol fromString(String value) {
         ListenerProtocol[] items = ListenerProtocol.values();
         for (ListenerProtocol item : items) {
            if (item.toString().equalsIgnoreCase(value)) {
               return item;
            }
         }
         return new IllegalArgumentException("Cannot find an enum for " + value);
      }

      @Override
      public String toString() {
         return this.value;
      }
   }
```

**Comment by andreaturli** [comment]
the build failure looks related, see https://jclouds.ci.cloudbees.com/job/jclouds-labs-pull-requests/org.apache.jclouds.labs$azurecompute/1710/testReport/junit/org.jclouds.azurecompute.features/DeploymentApiMockTest/testCreateWindows/

File: azurecompute/src/main/java/org/jclouds/azurecompute/compute/options/AzureComputeTemplateOptions.java (hunk 1 above)
Comment (bostko) [inline azurecompute/src/main/java/org/jclouds/azurecompute/compute/options/AzureComputeTemplateOptions.java]: Thank you very much for the suggestion.
Looks good however I am more for the simplest approach.

**Comment by nakomis** [comment]
@andreaturli Are you happy to merge this without the ListenerProtocol you suggested, or would you like to see that implemented?

**Comment by andreaturli** [comment]
+1 merging

**Comment by andreaturli** [comment]
does it need to be backport to 2.0.x, @bostko ?

**Comment by andreaturli** [comment]
merged at [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/2fe3e0c1)

**Comment by bostko** [comment]
I prefer to put effort in supporting more windowsconfiguration parameters in the TemplateOptions rather than focusing on this bit.
I'd like to pick this simple change in 2.0.x as well. Thanks.

**Comment by andreaturli** [comment]
makes sense please remember to open a jira issue to track the improvements you want to make.

----

merged at [2.0.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/69ef8765)

**Pull request closed without merging**